src/main/topLevelPages/myDashboard/newVirtualContributorWizard/CreateExternalAIDialog.tsx

```
@@ -0,0 +1,110 @@
+    assistantId: '',
+  };
+
+  const validationSchema = yup.object().shape({
+    engine: yup
+      .string()
+      .oneOf(PROVIDERS.map(({ id }) => id))
+      .required(),
+    apiKey: yup.string().required(),
+    assistantId: yup.string().when('engine', {
+      is: AiPersonaEngine.OpenaiAssistant,
+      then: yup.string().required(),
```

Biome

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)